build/server/konnectors/digiposte.js

```diff
@@ -0,0 +1,230 @@
+        },
+        json: true
+      }).then(function (folder) {
+        result.docs = folder.documents.filter(function (doc) {
+          return doc.invoice;
+        }).map(function (doc) {
+          var resultDoc = {
+            docid: doc.id,
+            type: doc.category,
+            date: moment(doc.creation_date),
+            pdfurl: 'https://secure.digiposte.fr/rest/content/document?_xsrf_token=' + xsrfToken,
+            amount: doc.invoice_data.chargeable_amount,
+            vendor: doc.sender_name
+          };
+          return resultDoc;
+        });
+        connector.logger.info('' + result.docs.length + ' bill(s)');
+        return result;
+      });
+    });
+  }).then(function (folders) {
+    // Then, for each folder, filter existing documents
+    connector.logger.info('Filtering existing documents');
+    return bb.each(folders, function (folder) {
+      return new Promise(function (resolve, reject) {
+        var entries = { fetched: folder.docs };
+        filterExisting(connector.logger, Bill)(requiredFields, entries, {}, function (err) {
+          if (err) {
+            connector.logger.error(err);
+          } else {
+            folder.docs = entries;
+          }
+          resolve();
+        });
+      });
+    }).then(function () {
+      return Promise.resolve(folders);
+    });
+  }).then(function (folders) {
+    return bb.each(folders, function (folder) {
+      connector.logger.info('Getting vendor ' + folder.name);
+      return new Promise(function (resolve, reject) {
+        saveDataAndFile(connector.logger, Bill, {
+          vendor: folder.name,
+          dateFormat: 'YYYYMMDD',
+          requestoptions: {
+            jar: j,
+            headers: {
+              'User-Agent': 'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:36.0) ' + 'Gecko/20100101 Firefox/36.0'
+            },
+            method: 'POST',
+            form: function form(doc) {
+              return {
+                'document_ids[]': doc.docid
+              };
+            }
+          }
+        }, ['bill'])({ folderPath: requiredFields.folderPath + '/' + folder.name }, folder.docs, {}, function (err) {
+          if (err) {
+            connector.logger.error('Error getting one of ' + JSON.stringify(folder.docs));
+          }
+          resolve();
+        });
+      });
+    }).then(function () {
+      return Promise.resolve(folders);
+    });
+  }).then(function (folders) {
+    // generation of the notification content
+    var nb = folders.reduce(function (memo, folder) {
+      if (folder.docs.filtered) memo += folder.docs.filtered.length;
+      return memo;
+    }, 0);
+    connector.logger.info('Number of fetched documents : ' + nb);
+    if (nb > 0) {
+      var localizationKey = 'notification bills';
+      var options = {
+        smart_count: nb
+      };
+      next(null, localization.t(localizationKey, options));
+    }
+  }).catch(function (err) {
+    next(err);
```

Comment:
That would be better to log the error and return the actual locale key corresponding to the error. Ex: 'bad credentials'.